### PR TITLE
[Docs] add a guide for adapter merging.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Content reused from https://huggingface.co/blog/peft_merging. Adapter merging is very popular in the diffusion community. Even though it's experimental, I think having it inside our docs will be beneficial.

### CODE REVIEW

File: docs/source/en/tutorials/adapter_merging.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+prompt = "toy_face of a hacker with a hoodie, pixel art"
+image = pipe(prompt, num_inference_steps=30, generator=torch.manual_seed(0)).images[0]
+image
+```
+
+![toy_face_pixel_art](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/peft_merging/toy_face_pixel_art.png)
+
+There are more examples of different merging methods available in [this post](https://huggingface.co/blog/peft_merging). We encourage you to give them a try 🤗
+
+## AnimateDiff
```
Comment (sayakpaul) [inline docs/source/en/tutorials/adapter_merging.md:194]: @DN6 would you like to give this a try? I think that would be very cool!

**Review by stevhliu — COMMENTED** [review]
Oh nice, I was wondering if it'd be possible to have `add_weighted_adapter` work with Diffusers! 🚀 

The only concern for me is I feel like merging/combining LoRA adapters is too spread out in the docs now ([here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_7110/en/using-diffusers/loading_adapters#-peft), [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_7110/en/tutorials/using_peft_for_inference), and [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_7110/en/tutorials/adapter_merging)). There are several options (I put a ⭐  for the one I'd recommend), so let me know what you think:

1. add the `add_weighted_adapter` method in the [Load adapters](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_7110/en/using-diffusers/loading_adapters#-peft) section
2. add the `add_weighted_adapter` method in the [Inference with PEFT](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_7110/en/tutorials/using_peft_for_inference) tutorial
3. create a new Merge LoRA guide in the Techniques section with `set_adapter` and `add_weighted_adapter` and then clean up the Load adapters doc to only show how to load ⭐

**Comment by sayakpaul** [comment]
@stevhliu gentle ping on the above.

**Comment by stevhliu** [comment]
Sorry for the delay!

Since it's experimental, I especially don't think it belongs in the Tutorials because I don't think we want to provide an unstable (meaning things can still change) foundation for new users learning how to merge LoRAs. I also think it's not great to introduce two ways of merging LoRAs in the Tutorials because it places extra cognitive work on the learner to choose/distinguish which method to use. We should focus on recommending one general approach in the Tutorial, and at the end link to the other option (whether its `add_weighted_adapter` or `set_adapters`).

Let me know what you think! 🙂

**Comment by sayakpaul** [comment]
Let’s go with option 3 then. Would you be able to repurpose this PR for the same?

**Comment by stevhliu** [comment]
Let me start a new PR, it'll be easier for me to refactor everything :)

**Comment by sayakpaul** [comment]
Sure. Feel free to re-purpose this one. Once that's opened, I will close this one.

**Comment by sayakpaul** [comment]
Closing in light of https://github.com/huggingface/diffusers/pull/7213.

**Pull request closed without merging**